web-api/src/personalization/routes.rs

```diff
@@ -820,6 +868,102 @@ async fn semantic_search(
     }))
 }
 
+#[instrument(skip(state, storage))]
+async fn recommendations(
+    state: Data<AppState>,
+    Json(body): Json<UnvalidatedRecommendationRequest>,
+    TenantState(storage): TenantState,
+) -> Result<impl Responder, Error> {
+    // TODO: actually return non-empty warnings in the response
+    let mut warnings = Vec::new();
+    let request = body
+        .validate_and_resolve_defaults(&state.config, &storage, &mut warnings)
+        .await?;
+
+    _recommendations(state, request, storage).await
+}
+
+async fn _recommendations(
+    state: Data<AppState>,
+    request: RecommendationRequest,
+    storage: Storage,
+) -> Result<impl Responder, Error> {
+    let RecommendationRequest {
+        count,
+        personalize,
+        include_properties,
+        include_snippet,
+        filter,
+        is_deprecated,
+    } = request;
+
+    let time = Utc::now();
+    let exclusions = personalized_exclusions(&storage, state.config.as_ref(), &personalize).await?;
+
+    let (interests, tag_weights) = match personalize.user {
+        InputUser::Ref { id } => {
+            storage::Interaction::user_seen(&storage, &id, time).await?;
+            (
+                storage::Interest::get(&storage, &id).await?,
+                storage::Tag::get(&storage, &id).await?,
+            )
+        }
+        InputUser::Inline { history } => {
+            let history = trim_history(
+                history,
+                state.config.personalization.max_stateless_history_for_cois,
+            );
+            let history = load_history(&storage, history).await?;
+            derive_interests_and_tag_weights(&state.coi, &history)
+        }
+    };
+
+    if interests.len() < state.coi.config().min_cois() {
+        return Ok(Either::Left((
+            deprecate!(if is_deprecated {
+                Json(PersonalizedDocumentsError::NotEnoughInteractions)
+            }),
+            StatusCode::CONFLICT,
+        )));
+    }
+
+    let mut documents = knn::CoiSearch {
+        interests: &interests,
+        excluded: &exclusions,
+        horizon: state.coi.config().horizon(),
+        max_cois: state.config.personalization.max_cois_for_knn,
+        count,
+        num_candidates: state.config.personalization.max_number_candidates,
+        time,
+        include_properties,
+        include_snippet,
+        filter: filter.as_ref(),
+    }
+    .run_on(&storage)
+    .await?;
+
+    rerank(
+        &state.coi,
+        &mut documents,
+        &interests,
+        &tag_weights,
+        state.config.semantic_search.score_weights,
```

Comment:
:wrench:  This is using the wrong config option.

```suggestion
        state.config.personalization.score_weights,
```

Reply:
done